### PR TITLE
[Refactor]: Refactor layer wise lr decay

### PR DESCRIPTION
**Pull request by YuanLiuuuuuu**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

**Before PR**:

- [ ] Pre-commit or other linting tools are used to fix the potential lint issues.
- [ ] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects, like MMDet or MMSeg.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: mmselfsup/engine/optimizers/layer_decay_optim_wrapper_constructor.py [hunk 1]
```diff
@@ -88,7 +90,6 @@ def add_params(self,
                 submodule of DCN, `is_dcn_module` will be passed to
```
Comment (fangyixiao18) [inline mmselfsup/engine/optimizers/layer_decay_optim_wrapper_constructor.py]: please update the docstring according to the args, `is_dcn_module` is removed

File: mmselfsup/engine/optimizers/layer_decay_optim_wrapper_constructor.py [hunk 2]
```diff
@@ -65,14 +65,16 @@ class LearningRateDecayOptimWrapperConstructor(DefaultOptimWrapperConstructor):
     """Different learning rates are set for different layers of backbone.
 
     Note: Currently, this optimizer constructor is built for ViT and Swin.
+
+    In addition to applying layer-wise learning rate decay schedule, this
+    module will not apply weight decay to ``normalization parameters``,
+    ``bias``, ``position embedding``, ``class token``, and
+    ``relative position bias table, automatically. What's more, the
+    ``paramwise_cfg`` in the base module will be ignored.
```
Comment (fangyixiao18) [inline mmselfsup/engine/optimizers/layer_decay_optim_wrapper_constructor.py:73]: Shall we add warnings if there are some settings in `paramwise_cfg`?

**Review by fangyixiao18 — CHANGES_REQUESTED** [review]

File: mmselfsup/engine/optimizers/layer_decay_optim_wrapper_constructor.py [hunk 3]
```diff
@@ -113,8 +117,11 @@ def add_params(self,
         for name, param in module.named_parameters():
             if not param.requires_grad:
                 continue  # frozen weights
+            # will not decay normalization params, bias, position embedding
+            # class token, relative position bias table
             if len(param.shape) == 1 or name.endswith('.bias') or name in (
-                    'backbone.pos_embed', 'backbone.cls_token'):
+                    'backbone.pos_embed', 'backbone.cls_token'
+            ) or name.endswith('.relative_position_bias_table'):
```
Comment (tonysy) [inline mmselfsup/engine/optimizers/layer_decay_optim_wrapper_constructor.py:126]: Please add UT for relative_position_bias_table

**Review by fangyixiao18 — APPROVED** [review]

**Review by tonysy — APPROVED** [review]
LGTM

**Pull request merged**